docs/getting-started/architecture-and-components/write-patterns.md

```diff
@@ -0,0 +1,68 @@
+# Writing Data to Feast
+
+Feast uses a [Push Model](getting-started/architecture-and-components/push-vs-pull-model.md) to push features to the online store.
+
+This has two important consequences: (1) communication patterns between the data producer and Feast and (2) feature computation and 
+_feature value_ write patterns to Feast's online store.
+
+Data Producers (i.e., services that generate data) send data to Feast so Feast can write it to the online store. That data can
+be either raw data where Feast computes and store stores the feature values or precomputed feature values.
+
+## Communication Patterns
+
+There are two ways to *_send_* data to the online store: 
+
+1. Synchronously
+   - Using an API call for a small number of entities or a single entity
+2. Asynchronously 
+   - Using an API call for a small number of entities or a single entity
+   - Using a "batch job" for a large number of entities
+
+Note, in some contexts, developers may "batch" a group of entities together and write them to the online store in a 
+single API call. This is a common pattern when writing data to the online store to reduce write loads but we would 
+not qualify this as a batch job.
+
+## Feature Value Write Patterns
+
+Writing feature values to the online store can be done in two ways: Precomputing the transformations or Computing the transformations On Demand. 
+
+### Combining Approaches
+
+In some advanced scenarios, a combination of precomputed and On Demand transformations might be optimal. For example, base feature calculations that do not change often could be precomputed and stored, while more dynamic calculations based on real-time data can be computed on demand. This hybrid approach can help balance the load on compute resources while ensuring feature freshness where it matters most.
+
+When selecting a feature value write pattern, consider the specific requirements of your application, such as the need for real-time data, the acceptable level of latency, and the computational resources available. Making an informed choice can significantly enhance the performance and reliability of your feature store operations.
+
+
+There are two ways to write *feature values* to the online store:
+
+1. Precomputing the transformations
+2. Computing the transformations On Demand
+3. Hybrid (Precomputed + On Demand)
+
+### 1. Precomputing the transformations
+Precomputed transformations can happen outside of Feast (e.g., via some batch job or streaming application) or inside of the Feast feature server when writing to the online store via the `write-to-online-store` api. 
```

Comment:
How can one precompute values during `write-to-online-store`? Are you referring to stream feature views?

Reply:
sure, but my understanding from that ticket and the discussion on the call was that we were talking about "persisting" values during get_online_features, not during write_to_online_store.

Reply:
>Again the write_to_online_store: bool parameter would dictate whether this ODFV would write to the online store and the force_compute: bool would dictate whether the ODFV would always recalculate the features. There's an argument to be made that we could skip the write_to_online_store in the FeatureView declaration but this metadata would be useful to have in the registry for users.

The ODFV computations were meant to happen when *writing* to the online store, as outlined in this document. 

I should have written this document first, I suppose.

Let me know if that makes more sense for you.